src/robot/libraries/String.py

```diff
@@ -229,13 +256,19 @@ def format_string(self, template, /, *positional, **named):
                 template = reader.read()
         return template.format(*positional, **named)
 
-    def get_line_count(self, string):
+    def get_line_count(self, string: "str | bytes") -> int:
         """Returns and logs the number of lines in the given string."""
         count = len(string.splitlines())
         logger.info(f"{count} lines.")
         return count
 
-    def split_to_lines(self, string, start=0, end=None):
+    # does not work with bytes: ${{b'hello\nworld'}} -> Length of '["b'hello\\nworld'"]' should be 2 but is 1.
+    def split_to_lines(
+        self,
+        string: str,
+        start: "int | str | None" = 0,
+        end: "int | str | None" = None,
```

Comment:
This is one of the keywords that are affected by #5558. Please use `Literal['']` instead of `str` with `start` and `end`.